docker/utils/context.py

```diff
@@ -0,0 +1,206 @@
+import os
+import re
+import tarfile
+from collections import namedtuple
+from .utils import lazy_line_reader
+
+lzma = None
+try:
+    import lzma
+except ImportError as ie:
+    try:
+        import backports.lzma as lzma
+    except ImportError:
+        pass
+
+
+class ContextError(Exception):
+    def __init__(self, msg):
+        self.message = msg
+
+
+INVALID_CONTEXT_FORMAT_LONG_FMT = """
+Build context at %s is not supported by docker-compose\\n
+The path must point to either:
+\t * A readable directory containing a valid Dockerfile
+\t * A tarball (optionally compressed with gzip, xz or bzip2)
+\t * A valid Dockerfile
+\t * A valid URL for a remote build context.
+%s"""
+
+INVALID_FIRST_INSTRUCTION_FMT = """
+Invalid first instruction in Dockerfile %s: '%s'\\n
+The first instruction in a Dockerfile must be "FROM ..."""
+
+# these prefixes are treated as remote by the docker daemon
+# (ref: pkg/urlutil/*) as of v1.6.0
+REMOTE_CONTEXT_PREFIXES = ["http://",
+                           "https://",
+                           "git://",
+                           "git@",
+                           "github.com/"]
+
+DOCKERFILE_CMD_RG = re.compile(r"[\t\s]*\\n$|([\s\t]*"
+                               r"FROM|MAINTAINER|RUN|CMD|EXPOSE|ENV|ADD|COPY|"
+                               r"ENTRYPOINT|VOLUME|USER|WORKDIR|ONBUILD"
+                               r"[\s\t]+.+)")
```

Comment:
Removed now. It would eventually be used for validating a Dockerfile thoroughly, but I discarded the idea.